modules/gdscript/gdscript_analyzer.cpp

```diff
@@ -3159,6 +3159,13 @@ void GDScriptAnalyzer::reduce_subscript(GDScriptParser::SubscriptNode *p_subscri
 		reduce_identifier(static_cast<GDScriptParser::IdentifierNode *>(p_subscript->base), true);
 	} else {
 		reduce_expression(p_subscript->base);
+
+		if (p_subscript->base->type == GDScriptParser::Node::ARRAY) {
+			GDScriptParser::ArrayNode *array = static_cast<GDScriptParser::ArrayNode *>(p_subscript->base);
```

Comment:
Any reason for this to use an extra variable while the other cases don't? It should work the same but I find this inconsistency odd, and probably so will the future people reading this code.